### ISSUE TITLE
Better "no docs found" error message

### ISSUE THREAD

**Issue by baggepinnen** [JuliaDocs/Documenter.jl#2334]
Whenever I get the error message that reads 
```
Error: no doc found for reference '[`rms`](@ref)' in src/validation.md.
```
I can never remember if the problem is
1. There is no docstring in the source code for this reference.
2. There is no docstring included in the docs for this reference.
3. Number 2 as a consequence of number 1.
4. Number 2 as a consequence of `rms` not being exported.

Would it be possible to somehow be more precise in these error messages? For example, I was just debugging why I got the message above, and it turned out that 
1. I did have `rms` in a `@docs` block.
2. `rms` did have a docstring in the source
3. `rms` _was not exported_.

It took me quite a while to figure out that the problem was number 3 here, and it's not the first time :sweat_smile:

**Comment by mortenpi** [comment]
Yep, it would definitely be possible for Documenter to be smarter here. The necessary information should all be there. Just needs to be extracted.

**Comment by uncomfyhalomacro** [comment]
Should Documenter add that to the documentation even if it is not exported as long as it is documented? 

Somewhat offtopic:
Idk if the recent merged PR for the public keyword can change how Documenter works. It seems likely. Will it still generate documentation for "private" API (those unexported)? Is there already a discussion about this somewhere in Documenter?

**Comment by mortenpi** [comment]
> Should Documenter add that to the documentation even if it is not exported as long as it is documented?

That's how Documenter works, yes. In light of `public`, we probably need an alternative mode that takes that metadata into account.
